include/nbl/asset/filters/kernels/CDiracImageFilterKernel.h

```diff
@@ -1,41 +1,32 @@
-#ifndef __NBL_ASSET_C_DIRAC_IMAGE_FILTER_KERNEL_H_INCLUDED__
-#define __NBL_ASSET_C_DIRAC_IMAGE_FILTER_KERNEL_H_INCLUDED__
+#ifndef _NBL_ASSET_C_DIRAC_IMAGE_FILTER_KERNEL_H_INCLUDED_
+#define _NBL_ASSET_C_DIRAC_IMAGE_FILTER_KERNEL_H_INCLUDED_
 
-#include "nbl/asset/filters/kernels/IImageFilterKernel.h"
-
-#include <ratio>
 #include <limits>
 
 namespace nbl::asset
 {
 
-class CDiracImageFilterKernel : public CFloatingPointIsotropicSeparableImageFilterKernelBase<CDiracImageFilterKernel>
+struct CDiracFunction
 {
-	using Base = CFloatingPointIsotropicSeparableImageFilterKernelBase<CDiracImageFilterKernel>;
-
-public:
-	CDiracImageFilterKernel() : Base(0.f) {}
+	constexpr static inline uint32_t k_smoothness = 0;
+	constexpr static inline float min_support = std::nextafter<float>(0.f,-1.f);
+	constexpr static inline float max_support = std::nextafter<float>(0.f,+1.f);
 
-	template <uint32_t derivative = 0>
-	inline float weight(float x, int32_t channel) const
+	template<int32_t derivative=0>
```

Comment:
I don't get the reason of making it `int32_t` from `uint32_t`. It doesn't matter for positive values because we probably won't have so much higher order derivatives, but do you expect `derivative` to be negative?

Reply:
well if its negative you'd integrate, but yeah it doesn't matter much